src/shared/sam/cli/samCliLocator.ts

```diff
@@ -69,8 +72,37 @@ abstract class BaseSamCliLocator {
 
         for (const fullPath of fullPaths) {
             this.logger.verbose(`Searching for SAM CLI in: ${fullPath}`)
+            const context: SamCliValidatorContext = {
+                samCliLocation: async () => fullPath,
+                getSamCliExecutableId: async () => 'temp',
+                getSamCliInfo: async () => {
+                    const samCliInfo = new SamCliInfoInvocation(
+                        new DefaultSamCliProcessInvoker({
+                            cliConfig: {
+                                getOrDetectSamCli: async () => {
+                                    return { path: fullPath, autoDetected: false }
+                                },
+                            },
+                        })
+                    )
+
+                    return await samCliInfo.execute()
+                },
+            }
+            const validator = new DefaultSamCliValidator(context)
             if (await filesystemUtilities.fileExists(fullPath)) {
-                return fullPath
+                try {
+                    const validationResult = await validator.getVersionValidatorResult()
+                    if (validationResult.validation === SamCliVersionValidation.Valid) {
+                        return fullPath
+                    } else {
+                        this.logger.info(
+                            `Detected SAM executable at ${fullPath} invalid: ${validationResult.validation}`
+                        )
+                    }
+                } catch (e) {
+                    /*swallow err*/
```

Comment:
Unless you expect this to be very noisy (lots of unavoidable, expected errors...?) we should never swallow without at least logging something.

Reply:
Error in this case would be an executable that doesn't work. Is there value in having that around? I think `getVersionValidatorResult` also throws an error.

Reply:
can `e` be checked and only skipped if it looks like a "executable failed" error? Then only log non-matching errors.